src/main/java/data_structures/LinkedList/LinkedList.java

```diff
@@ -39,10 +42,30 @@ public boolean includes(int value) {
         return false;
     }
     //return collection of all current node values.
-    public void print() {
+    public List<Integer> print() {
         Node currentNode = this.head;
+        List<Integer> list = new ArrayList<>();
         while (currentNode != null) {
             System.out.println(currentNode.value);
+            list.add(currentNode.value);
+            currentNode = currentNode.next;
+        }
+        System.out.println(list);
+        return list;
+    }
+
+    //code challenge 07 k-th from end
+    public <T>  nullIndexSearch(int k, <T> linkedListGeorge) {
+        <T> current = linkedListGeorge.head;
+        ArrayList <T>  linkedListArrayList = new ArrayList<>();

```

Comment:
I don't know why you use an arraylist here. You should create a method size() on this LinkedList Class that returns the size of the list.